restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java

```diff
@@ -94,18 +110,6 @@ public JsonNode convertToJsonDoc(
         JsonNode childRef;
 
         if (isArray && !writeAllPathsAsObjects) {
-          if (!ref.isArray()) {
```

Comment:
This is dead code, I added it as a safeguard long ago but it's never actually hit